src/server.js

```diff
@@ -13,25 +22,31 @@ MeteorX.onReady = function(cb) {
 };
 
 MeteorX.Server = Meteor.server.constructor;
+
+installMongoDetector(MeteorX);
+
 exposeLivedata(MeteorX);
 
-// before using any other MeteorX apis we need to hijack Mongo related code
-// that'w what we are doing here.
-Meteor.startup(function() {
-  runWithAFiber(function() {
+function initSync() {
+  runWithAFiber(() => {
     exposeMongoLivedata(MeteorX);
   });
 
-  MeteorX._readyCallbacks.forEach(function(fn) {
-    runWithAFiber(fn);
-  });
+  MeteorX._readyCallbacks.map(runWithAFiber);
   MeteorX._ready = true;
-});
+}
+
+async function initAsync() {
+  await exposeMongoAsync(MeteorX);
 
-function runWithAFiber(cb) {
-  if (Fibers.current) {
-    cb();
-  } else {
-    new Fiber(cb).run();
+  for (const cb of MeteorX._readyCallbacks) {
+    await cb();
   }
+
+  MeteorX._ready = true;
 }
```

Comment:
I'm wondering if we can combine these two functions without adding much complexity:
```js
async function init() {
  if (fibersEnabled) {
    exposeMongoLivedata(MeteorX);
  } else {
    await exposeMongoAsync(MeteorX);
  }

 for (const cb of MeteorX._readyCallbacks) {
     await cb();
  }

  MeteorX._ready = true;
}
```

Before Meteor 3, async functions in packages/app code always run within a fiber so we don't have to manually wrap it ourselves

If we want to make startup wait for this async function, we could do something like this:
```
Meteor.startup(() => {
  let promise = init();
  if (promise.await) promise.await();

  return promise;
});
```

Reply:
I don't think it is a good idea to combine them, having them separate makes it way easier to understand and maintain, even if we have some duplicated code.